cocotb/regression.py

```diff
@@ -201,13 +204,20 @@ def _discover_tests() -> Iterable[Test]:
 
                     yield test
 
-                # only look in first module for all functions and don't complain if all functions are not found
-                break
+                # Clear not_found_tests for the next module search
+                tests = not_found_tests.copy()
+                not_found_tests.clear()
```

Comment:
```suggestion
                # use not_found_tests for the next module search
                tests = not_found_tests
```

Reply:
Changed in 0d656990c6bc950012502e9d25ff767c6ea59468.